// @strapi/upload/dist/admin/future/services/assets.mjs
import { uploadApi } from './api.mjs';

const assetsApi = uploadApi.injectEndpoints({
    endpoints: (builder)=>({
            getAssets: builder.query({
                query: (params = {})=>{
                    const { folder, ...rest } = params;
                    const queryParams = {
                        ...rest
                    };
                    if (folder != null) {
                        queryParams['filters'] = {
                            $and: [
                                {
                                    folder: {
                                        id: folder
                                    }
                                }
                            ]
                        };
                    } else {
                        queryParams['filters'] = {
                            $and: [
                                {
                                    folder: {
                                        id: {
                                            $null: true
                                        }
                                    }
                                }
                            ]
                        };
                    }
                    return {
                        url: '/upload/files',
                        method: 'GET',
                        config: {
                            params: queryParams
                        }
                    };
                },
                transformResponse: (response)=>response,
                providesTags: (result)=>result ? [
                        ...result.results.map(({ id })=>({
                                type: 'Asset',
                                id
                            })),
                        {
                            type: 'Asset',
                            id: 'LIST'
                        }
                    ] : [
                        {
                            type: 'Asset',
                            id: 'LIST'
                        }
                    ]
            })
        })
});
const { useGetAssetsQuery } = assetsApi;

export { useGetAssetsQuery };
